### PR TITLE
Feature: optimise example video

### PR DESCRIPTION
**Pull request by gozmanyoni**
Optimizing example - removed unneeded transformations and methods and further simplified code.
From testing on Pi 4 60FPS its usable with 40-65% cpu, before these changes CPU was 50-80%

@steelbrain @rhysmorgan134 curious if you have a similar experience on this branch?

### CODE REVIEW

**Review by steelbrain — COMMENTED** [review]
Changeset makes sense. Thank you for working on it!

I’ll try it this evening and report back

**Comment by rhysmorgan134** [comment]
I've done some testing

new changes 

Screenshot 2023-10-06 at 21 01 01

current master branch

Screenshot 2023-10-06 at 21 01 05

jmuxer/electron

Screenshot 2023-10-06 at 21 10 03

Not particuarly useful as it's obviously a 1min average, but test was to go to map, choose navigate to work, let maps zoom in, then manually zoom out 5 clicks, then cancel route.

Worth noting, I am unsure how jmuxer scored so high, jmuxer @ 1080 @ 60fps feels the worst in the case of useablility but got the lowest averages, I am guessing it is just dropping frames so technically lowering the load, but causing severe lag.

I would say the new changes feel better, but I still think 1080 @ 60fps on all implementations doesn't feel "smooth". This is all with gpu mem set to 256mb on a pi4 4gb

**Comment by gozmanyoni** [comment]
Added more renderers - particularly interested in WebGPU (supposed to be faster than WebGL), which we need Vulkan for - planning to try and enable Vulkan on the pi following - https://qengineering.eu/install-vulkan-on-raspberry-pi.html

You can also enable fps reporting to see console logs with FPS every 5 seconds

File: examples/carplay-web-app/src/App.tsx [hunk 1]
```diff
@@ -26,10 +26,12 @@ const config: Partial<DongleConfig> = {
   width,
   height,
   fps: 60,
-  mediaDelay: 0,
+  mediaDelay: 300,
```
Comment (gozmanyoni) [inline examples/carplay-web-app/src/App.tsx:29]: Was configuring the dongle in its own web interface, and it actually errors out when sending anything below 300 - aligning with the minimum there.

**Comment by steelbrain** [comment]
> Added more renderers - particularly interested in WebGPU (supposed to be faster than WebGL), which we need Vulkan for - planning to try and enable Vulkan on the pi following - https://qengineering.eu/install-vulkan-on-raspberry-pi.html

The referenced blog post seems very heavy weight just to try Vulkan, here’s something from ChatGPT https://chat.openai.com/share/bbaefbb3-9115-41c1-9cf9-0d4f3086dc53

Looks like it’s bundled in raspi repositories now?

**Comment by rhysmorgan134** [comment]
Looks good to me, will give the different renderers a try when I get a chance!

**Review by rhysmorgan134 — APPROVED** [review]

**Review by steelbrain — APPROVED** [review]
Alright, finally tried all the renderers, they all work flawlessly. Thank you @gozmanyoni!

**Comment by steelbrain** [comment]
@gozmanyoni It was on macOS, I spent a few minutes on the pi and gave up after Chromium didnt enable webgpu even after vulkan drivers were installed.

I can try again on the weekend and document what works and what doesnt

**Comment by rhysmorgan134** [comment]
@gozmanyoni @steelbrain silky smooth on a pi5 even at 1080@60fps!

**Comment by gozmanyoni** [comment]
@rhysmorgan134 nice! Do you have a demo video? 

Also - does the pi5 play well with PiMOST? No stock of these in Australia yet :-(

**Comment by rhysmorgan134** [comment]
@gozmanyoni no video yet, will grab one though.

Pi5 itself seems to be working well with it, however bookworm is having issues with i2s and the simple-audio-card overlay!

**Pull request merged**